framework/source/class/qx/ui/window/Window.js

```diff
@@ -372,6 +372,96 @@ qx.Class.define("qx.ui.window.Window",
       check : "Boolean",
       init : false,
       apply : "_applyShowStatusbar"
+    },
+
+
+
+
+    /*
+    ---------------------------------------------------------------------------
+      WHEN TO AUTOMATICALY CENTER
+    ---------------------------------------------------------------------------
+    */
+
+    /** 
+     * When should the window be automatically centered
+     * 
+     * This property value is formed from the following strings.
+     *
+     *   "appear" - Center the window when it appears (or reappears)
+     *   "resize" - Center the window when its parent container is resized
+     * 
+     * An empty array removes all automatic centering.
+     * 
+     * Otherwise, the property value should be an array containing one or more
+     * of the above strings.
+     * 
+     * Examples:
+     *   Center upon appear and resize:
+     *     win.setCenterWhen([ "appear", "resize" ]);
+     * 
+     *   Center only upon appear:
+     *     win.setCenterWhen([ "appear" ]);
+     * 
+     *   Remove automatic centering:
+     *     win.setCenterWhen([]);
+     */
+    centerWhen :
+    {
```

Comment:
I'm here with @cajus: It would be much simpler and understandably if we had two boolean properties ```centerOnAppear``` and ```centerOnResize```. No arrays, no initialization hassle.